db/db_impl/db_impl.cc

```diff
@@ -744,8 +782,13 @@ Status DBImpl::GetStatsHistory(
   if (!stats_iterator) {
     return Status::InvalidArgument("stats_iterator not preallocated.");
   }
-  stats_iterator->reset(
-      new InMemoryStatsHistoryIterator(start_time, end_time, this));
+  if (immutable_db_options_.persist_stats_to_disk) {
```

Comment:
Probably not necessarily in this PR. In the future, it would be nice if we can have `StatsHistoryIterator` unifying the in-memory, on-disk case. Or we can get rid of the in-memory case.